src/ui/components/results/appliedfilterscomponent.js

```diff
@@ -0,0 +1,156 @@
+      resetPagination: true,
+      useFacets: true
+    });
+  }
+
+  /**
+   * Returns the currently applied nlp filter nodes, with nlp filter nodes that
+   * are duplicates of other filter nodes removed or filter on hiddenFields removed.
+   * @returns {Array<FilterNode>}
+   */
+  _getPrunedNlpFilterNodes () {
+    const duplicatesRemoved = this.nlpFilterNodes.filter(nlpNode => {
+      const isDuplicate = this.appliedFilterNodes.find(appliedNode =>
+        appliedNode.hasSameFilterAs(nlpNode)
+      );
+      return !isDuplicate;
+    });
+    return pruneFilterNodes(duplicatesRemoved, this._config.hiddenFields);
+  }
+
+  /**
+   * Combine all of the applied filters into a format the handlebars
+   * template can work with.
+   * Keys are the fieldName of the filter. Values are an array of objects with a
+   * displayValue and dataFilterId.
+   * @returns {Array<Object>}
+   */
+  _groupAppliedFilters () {
+    const getFieldName = filterNode => filterNode.getMetadata().fieldName;
+    const parseNlpFilterDisplay = filterNode => ({
+      displayValue: filterNode.getMetadata().displayValue
+    });
+    const parseRemovableFilterDisplay = (filterNode, index) => ({
+      displayValue: filterNode.getMetadata().displayValue,
+      dataFilterId: index,
+      removable: this._config.removable
+    });
+    const removableNodes = groupArray(this.appliedFilterNodes, getFieldName, parseRemovableFilterDisplay);
+    const prunedNlpFilterNodes = this._getPrunedNlpFilterNodes();
+    return groupArray(prunedNlpFilterNodes, getFieldName, parseNlpFilterDisplay, removableNodes);
+  }
+
+  /**
+   * Returns an array of object the handlebars can understand and render
+   * the applied filters bar from. Our handlebars can only loop through arrays,
+   * not objects, so we need to reformat the grouped applied filters.
+   * @returns {Array<Object>}
+   */
+  _createAppliedFiltersArray () {
+    const groupedFilters = this._groupAppliedFilters();
+    return Object.keys(groupedFilters).map(label => ({
+      label: label,
+      filterDataArray: groupedFilters[label]
+    }));
+  }
+
+  /**
+   * Pulls applied filter nodes from {@link FilterRegistry}, then retrives an array of
+   * the leaf nodes, and then removes hidden or empty {@link FilterNode}s. Then appends
+   * the currently applied nlp filters.
+   */
+  _calculateAppliedFilterNodes () {
+    const filterNodes = this.core.filterRegistry.getAllFilterNodes();
+    const simpleFilterNodes = flattenFilterNodes(filterNodes);
+    return pruneFilterNodes(simpleFilterNodes, this._config.hiddenFields);
+  }
+
+  setState (data) {
+    const verticalResults = this.core.globalStorage.getState(StorageKeys.VERTICAL_RESULTS) || {};
+    const nlpFilters = verticalResults.appliedQueryFilters || [];
+
+    /**
+     * Array of nlp filters in the search response.
+     * @type {Array<AppliedQueryFilter>}
+     */
+    this.nlpFilterNodes = convertNlpFiltersToFilterNodes(nlpFilters || []);
```

Comment:
Is the `nlpFilters || []` fallback logic duplicated here and on the line above?

Reply:
ah yeah that's duplicated, removed the second fallback!